llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -471,6 +471,11 @@ Value *VPInstruction::generatePerPart(VPTransformState &State, unsigned Part) {
       assert(State.VF.isScalable() && "Expected scalable vector factor.");
       Value *VFArg = State.Builder.getInt32(State.VF.getKnownMinValue());
 
+      if (getNumOperands() == 3) {
+        Value *MaxSafeVF = State.get(getOperand(2), VPIteration(0, 0));
```

Comment:
Document ExplicitVectorLength opcode in VPlan.h wit additional optional operand)

Reply:
Will do